python/cudf/cudf/core/indexed_frame.py

```diff
@@ -325,12 +363,14 @@ def _from_data(
                 f"index must be a cudf.Index not {type(index).__name__}"
             )
         out._index = index
+        out._attrs = {} if attrs is None else attrs
         return out
 
     @_performance_tracking
     def _from_data_like_self(self, data: MutableMapping):
         out = super()._from_data_like_self(data)
         out.index = self.index
+        out._attrs = copy.deepcopy(self._attrs)
```

Comment:
question: Why can't we just shallow copy in this case?

Reply:
We will need a deep copy because we want to keep the attrs as new separate copies:
```

In [1]: import pandas as pd

In [2]: df = pd.DataFrame([[1, 2], [3, 4]], dtype="Int64")

In [3]: df.attrs = {"a":100}

In [4]: ndf = df + 10

In [5]: ndf
Out[5]: 
    0   1
0  11  12
1  13  14

In [6]: ndf.attrs
Out[6]: {'a': 100}

In [7]: ndf.attrs['a'] = 20

In [8]: ndf
Out[8]: 
    0   1
0  11  12
1  13  14

In [9]: ndf.attrs
Out[9]: {'a': 20}

In [10]: df.attrs
Out[10]: {'a': 100}
```